pkg/types/baremetal/platform.go

```diff
@@ -0,0 +1,66 @@
+package baremetal
+
+// BMC stores the information about a baremetal host's management controller.
+type BMC struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+	Address  string `json:"address"`
+}
+
+// Host stores all the configuration data for a baremetal host.
+type Host struct {
+	Name            string `json:"name,omitempty"`
+	BMC             BMC    `json:"bmc"`
+	Role            string `json:"role"`
```

Comment:
I can't decide how I feel about the "Role" being here. We could instead define the hosts under the `compute` and `controlPlane` platform sections.  What would happens when the replica count is not equal to the count of Hosts? Aren't hosts supposed to be generic? Shouldn't I (eventually) be able to use any unprovisioned host as a master or worker? Is the "Role" really an artifact of the host itself? Is it's silicon special?

Reply:
It could be special, if a user has a number of heterogeneous servers and wants particular ones dedicated to be masters.  I wouldn't want to move it under compute/controlPlane and specifically require that users declare a role, though.

I think once we have workers being deployed by the installer, we'll be rethinking some of this. I'd expect this to become optional, and if unspecified let Ironic pick a random bare metal host through the Allocation mechanism we use.

Reply:
Here's the thinking:

All defined hosts get turned into `BareMetalHost` objects.

Assuming we get more than the replica count for masters, the installer has to know how to choose some subset of hosts to provision as masters.  That's what the role was for.

True, we could move these under the controlPlane and compute sections and we wouldn't need the role anymore.  That's something to consider as a follow-up.  I figure all of this is still subject to change while it's experimental.

Reply:
Your thought is that in 4.2 this API is experimental, so I should be ok with it?  Wish we had some way to denote the instability of his one part of the installconfig :-(

Reply:
well, "accept it as experimental" more than "be ok with it"  :-)

I tried to document in `docs/user/metal/install_ipi.md` how the install-config is subject to change while this is still experimental.  Recommendations welcome on any other and better ways we could make this clear.

Reply:
Ohhh, less terrible idea.  A `[]string` in the `compute` and `controlPlane` bare metal platform block that has a list of names. And those names are the hosts here.  so all of the hosts are in 1 place, like this, but which one goes where is in the block where it most makes sense.

Reply:
Assuming I get a promise to re-evaluate this before 4.3/GA I'm ok with it how it stands. Though I don't know how it works if things don't match up. I guess that's a QE test case.

Reply:
TBH, I don't even know if the code even looks at `role` yet!  I don't think it does.  I'll file a bug on this now to follow up.

Reply:
I've filed https://github.com/openshift-metal3/kni-installer/issues/165 to keep track of this.

Reply:
https://github.com/openshift/installer/issues/2003